packages/mdl-snackbar/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDLComponent from 'mdl-base';
+import MDLSnackbarFoundation from './foundation';
+
+export {MDLSnackbarFoundation};
+
+export default class MDLSnackbar extends MDLComponent {
+  static buildDom() {
+    const {ROOT: CSS_ROOT, TEXT, ACTION} = MDLSnackbarFoundation.cssClasses;
+
+    const root = document.createElement('div');
+    root.classList.add(CSS_ROOT);
+    // TODO: Aria attributes for root?
+    root.innerHTML = `
+      <div class="${TEXT}"></div>
+      <button type="button" class="${ACTION}"></button>
+    `;
+
+    return root;
+  }
+
+  static attachTo(root) {
+    return new MDLSnackbar(root);
+  }
+
+  showSnackbar(data) {
+    this.foundation_.showSnackbar(data);
+  }
+
+  getDefaultFoundation() {
+    const {TRANS_END_EVENT_NAME, TEXT_SELECTOR, ACTION_SELECTOR} = MDLSnackbarFoundation.strings;
+    const nativeText = this.root_.querySelector(TEXT_SELECTOR);
+    const nativeAction = this.root_.querySelector(ACTION_SELECTOR);
+    /* eslint brace-style: "off" */
```

Comment:
Without this, it wants the `setActionText` and `setMessageText` callbacks to 1) Have braces to avoid an implicit return of a assignment expression, 2) because it has braces, it wants the statements formatted on their own line, etc.

IMHO, it results in a much less readable adapter implementation, so I disabled that check. I'm happy to go the other way if preferred.